dotCMS/src/main/java/com/dotcms/graphql/AbstractGraphQLHttpServlet.java

```diff
@@ -0,0 +1,446 @@
+import graphql.servlet.GraphQLInvocationInputFactory;
+import graphql.servlet.GraphQLMBean;
+import graphql.servlet.GraphQLObjectMapper;
+import graphql.servlet.GraphQLQueryInvoker;
+import graphql.servlet.GraphQLServletListener;
+import graphql.servlet.GraphQLSingleInvocationInput;
+import graphql.servlet.internal.GraphQLRequest;
+import graphql.servlet.internal.VariableMapper;
+
+/**
+ * @author Andrew Potter
+ */
+public abstract class AbstractGraphQLHttpServlet extends HttpServlet implements Servlet, GraphQLMBean {
+
+    public static final Logger log = LoggerFactory.getLogger(AbstractGraphQLHttpServlet.class);
+
+    public static final String APPLICATION_JSON_UTF8 = "application/json;charset=UTF-8";
+    public static final String APPLICATION_GRAPHQL = "application/graphql";
+    public static final int STATUS_OK = 200;
+    public static final int STATUS_BAD_REQUEST = 400;
+
+    private static final GraphQLRequest INTROSPECTION_REQUEST = new GraphQLRequest(IntrospectionQuery.INTROSPECTION_QUERY, new HashMap<>(), null);
+    private static final String[] MULTIPART_KEYS = new String[]{"operations", "graphql", "query"};
+
+    private GraphQLConfiguration configuration;
+
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    protected abstract GraphQLQueryInvoker getQueryInvoker();
+
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    protected abstract GraphQLInvocationInputFactory getInvocationInputFactory();
+
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    protected abstract GraphQLObjectMapper getGraphQLObjectMapper();
+
+    /**
+     * @deprecated override {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    protected abstract boolean isAsyncServletMode();
+
+    protected GraphQLConfiguration getConfiguration() {
+        return GraphQLConfiguration.with(getInvocationInputFactory())
+                .with(getQueryInvoker())
+                .with(getGraphQLObjectMapper())
+                .with(isAsyncServletMode())
+                .with(listeners)
+                .build();
+    }
+
+    /**
+     * @deprecated use {@link #getConfiguration()} instead
+     */
+    @Deprecated
+    private final List<GraphQLServletListener> listeners;
+
+    private HttpRequestHandler getHandler;
+    private HttpRequestHandler postHandler;
+
+    public AbstractGraphQLHttpServlet() {
+        this(null);
+    }
+
+    public AbstractGraphQLHttpServlet(List<GraphQLServletListener> listeners) {
+        this.listeners = listeners != null ? new ArrayList<>(listeners) : new ArrayList<>();
+    }
+
+    @Override
+    public void init(ServletConfig servletConfig) {
+        this.configuration = getConfiguration();
+
+        this.getHandler = (request, response) -> {
+            GraphQLInvocationInputFactory invocationInputFactory = configuration.getInvocationInputFactory();
```

Comment:
this looks such as strategy, one for the scheme, one for the batch and one for the other query.